lib/ogsf/gp3.c

```diff
@@ -295,5 +297,7 @@ int Gp_load_sites_thematic(geosite *gp, struct Colors *colors)
             _("%d points without category. "
               "Unable to determine color rules for features without category."),
             nskipped);
+    db_close_database_shutdown_driver(driver);
+    Vect_destroy_field_info(Fi);
```

Comment:
```suggestion
    if (driver) {
        db_close_database_shutdown_driver(driver);
        Vect_destroy_field_info(Fi);
    }
```

The db driver might not be available, see L202.

Also, the calls to `db_select_value()` above need to be protected with if statements if there is no database connection (lines 235ff).

Reply:
> Also, the calls to db_select_value() above need to be protected with if statements if there is no database connection (lines 235ff).

- If `Fi` fails to be set (L201ff), then `driver` is not set either: both remain NULL.
- If `Fi` is successfully set, but `driver` fails to open leads to G_fatal_error.

So,  both `Fi` and `driver` are either NULL or !NULL from L211.

With `driver` and `Fi` uninitiated (NULL), `db_select_value` returns  `-1` and continues the loop:

```c
            nvals = db_select_value(driver, Fi->table, Fi->key, cat,
                                    gp->tstyle->color_column, &value);
            if (nvals < 1)
                continue;
```

Checking for `driver` is not necessary as it works as is. There's no doubt this function may be improved for clarity, but that change – as far as I understand – is not needed for the initial aim of this PR.